### DISCUSSION TITLE
Do I interpret the actions log correctly?

### DISCUSSION THREAD

**Discussion by jsbien** [trufanov-nok/djvudict#2] [Q&A]
I assume that in the line `id: 0	x: 89	y: 600` x and y are coordinates of the `lib_00000.bmp` snippet. However when I convert it to an url such as `file:test.djvu?djvuopts=&highlight=00089,00600,00047,00050` the highlighting is shifted vertically several pixels. I enclose the test file just in case you want to have a look at it.
[test.zip](https://github.com/trufanov-nok/djvudict/files/13265295/test.zip)
P.S. My purpose is to visualize the dictionary with djview4poliqarp, more information will be soon available in a paper.

**Comment by trufanov-nok** [comment]
Hi, It seems so long I was't working with djvu so I might forget something. But it looks like:
1. Djvu coordinate axes starts at **botom** left corner
2. djvudict prints out in actions.log **top** left corner of the litera position (89, 600) in our case
3. size of the litera could be obtained from bmp file (width 50, height 47)
4. Then I may use djvu annotations to highlight it with djvulibre's djvused and following commands:
```
cd /tmp
echo '(maparea "" "test" (rect 89 553 50 47 )  (hilite #FFFF80 ) (opacity 80) )' > ann.txt
djvused test.djvu -e "select 1; set-ant ann.txt; save"
```
according to the specs rect must be "(rect xmin ymin width height)", notice **ymin** and width height order.

After that djview4 shows me pretty fine result:
![image](https://github.com/trufanov-nok/djvudict/assets/5746456/3e9bbe7d-595a-4329-ba4e-d17ce37b9554)

I'm not sure which app you're using to display djvu with highlights. Perhaps you or the app swap width and height order of the litera.
[test (1).zip](https://github.com/trufanov-nok/djvudict/files/13267799/test.1.zip)

**Reply by jsbien** [reply]
Thanks! Swaping width and height order was just my mistake. However I was not aware I have to calculate **bottom** left corner. I hope now it will work correctly. I wil close the discussion after testing.

**Comment by jsbien** [comment]
It works. Thank you very much!
